core/src/main/java/hudson/Util.java

```diff
@@ -1927,4 +1927,32 @@ public static long daysElapsedSince(@NonNull Date date) {
     private static PathRemover newPathRemover(@NonNull PathRemover.PathChecker pathChecker) {
         return PathRemover.newFilteredRobustRemover(pathChecker, DELETION_RETRIES, GC_AFTER_FAILED_DELETE, WAIT_BETWEEN_DELETION_RETRIES);
     }
+
+    /**
+     * Returns SHA-256 Digest of input bytes
```

Comment:
the `null` handling here is testing in a unit test so is considered defined.  so how can the input here be marked as `NonNull` when `null` is explicitly tested - either it can be null and should have the test or it can not be null and the test is unneeded.

the marking of `input` as `NonNull` should be generating a SpotBugs warning about a redundant check of the input - if it is not - then there is a bug in SpotBugs.